src/compiler/transformers/decorators-to-static/component-decorator.ts

```diff
@@ -6,6 +6,27 @@ import { convertValueToLiteral, createStaticGetter, retrieveTsDecorators } from
 import { getDeclarationParameters } from './decorator-utils';
 import { styleToStatic } from './style-to-static';
 
+/**
+ * Perform code generation to create new class members for a Stencil component
+ * which will drive the runtime functionality specified by various options
+ * passed to the `@Component` decorator.
+ *
+ * Inputs are validated (@see {@link validateComponent}) before code generation
+ * is performed.
+ *
+ * **Note**: in this function and in functions that it called the `newMembers`
+ * parameter is treated as a out parameter and mutated, with new class members
+ * added to it.
```

Comment:
Nit - this reads a little funky. Maybe something like:
```suggestion
 * **Note**: in this function and in functions that it calls, the `newMembers`
 * parameter is treated as an out parameter and mutated, with new class members
 * added to it.
```